internal/service/cluster/cluster.go

```diff
@@ -269,62 +299,16 @@ func (s *Service) Run(ctx context.Context, host service.Host) error {
 	ctx, cancel := context.WithCancel(ctx)
 	defer cancel()
 
-	limiter := rate.NewLimiter(rate.Every(stateUpdateMinInterval), 1)
-	s.node.Observe(ckit.FuncObserver(func(peers []peer.Peer) (reregister bool) {
-		tracer := s.tracer.Tracer("")
```

Comment:
This logic was moved below to a goroutine that handles dispatching cluster change notifications. This is because when cluster minimum size timer expires we also need to dispatch cluster change notifications, even though the peers in ckit didn't change.